### PR TITLE
Regression: `Admin UI -> Rooms` Search filter not working

### PR DESCRIPTION
**Pull request by totoi690**
## Issue Description
In Rocket.Chat it is possible to create channels with UTF-8 and non-Latin characters (Cyrillic or Greek for example) but the admin interface cannot find them.
The channel with Latin characters finds it well.

## Steps to Reproduce
1. Enable `UTF8 Names Validation` on `Admin UI → General`
2. Enable `Allow Special Characters in Room Names` on `Admin UI → Layout → User Interface`
3. Create a channel and private room with Cyrillic names, like `тестовый канал` and `тестовый чат` or Greek `Ελληνικά`
4. Go to Admin UI - Rooms and try to find that room and channel

## Expected Behavior
Non-Latin search should work in the `Admin → Rooms ` area and any other search field in Rocket.Chat interface even when the option `Allow Special Characters in Room Names` is enabled.

TC-385, TC-444, TC-499

### CODE REVIEW

File: apps/meteor/app/api/server/v1/rooms.js [hunk 1]
```diff
@@ -333,14 +333,15 @@ API.v1.addRoute(
 		get() {
 			const { offset, count } = this.getPaginationItems();
 			const { sort } = this.parseJsonQuery();
-			const { types, filter } = this.requestParams();
+			const { types, filter, useFname } = this.requestParams();
```
Comment (gabriellsh) [inline apps/meteor/app/api/server/v1/rooms.js:336]: This here is something I have some questions about.

Should we expect this value to come from the request, or should we get it directly from the settings? (even if we expect it to come from the request, I think we should get the setting and overwrite the value if it is sent through a parameter)

**Review by gabriellsh — CHANGES_REQUESTED** [review]
We can work on this together if you have questions. Please dm me if you need.

File: apps/meteor/server/models/raw/Rooms.js [hunk 2]
```diff
@@ -75,13 +75,16 @@ export class RoomsRaw extends BaseRaw {
 					},
 			  };
 
+		const fnameQuery = useFname ? [{ fname: nameRegex }] : [];
```
Comment (LucianoPierdona) [inline apps/meteor/server/models/raw/Rooms.js:78]: shouldn't it be like this?
```suggestion
		const fnameQuery = useFname ? { fname: nameRegex } : {};
```

File: apps/meteor/server/models/raw/Rooms.js [hunk 3]
```diff
@@ -128,10 +131,13 @@ export class RoomsRaw extends BaseRaw {
 
 		const onlyTeamsCondition = onlyTeams ? { $and: [{ teamMain: { $exists: true } }, { teamMain: true }] } : {};
 
+		const fnameQuery = useFname ? [{ fname: nameRegex }] : [];
```
Comment (LucianoPierdona) [inline apps/meteor/server/models/raw/Rooms.js:134]: same here
```suggestion
		const fnameQuery = useFname ? { fname: nameRegex } : {};
```

**Review by LucianoPierdona — CHANGES_REQUESTED** [review]
Small points, otherwise backend changes looks good

**Review by LucianoPierdona — APPROVED** [review]

**Review by hugocostadev — DISMISSED** [review]

File: apps/meteor/server/models/raw/Rooms.js [hunk 4]
```diff
@@ -62,7 +62,7 @@ export class RoomsRaw extends BaseRaw {
 		return statistic;
 	}
 
-	findByNameContainingAndTypes(name, types, discussion = false, teams = false, showOnlyTeams = false, options = {}) {
+	findByNameOrFnameContainingAndTypes(name, types, discussion = false, teams = false, showOnlyTeams = false, options = {}) {
```
Comment (rodrigok) [inline apps/meteor/server/models/raw/Rooms.js:65]: When you change a function name you need to change all the places that uses it

**Review by rodrigok — CHANGES_REQUESTED** [review]

File: apps/meteor/server/models/raw/Rooms.js (hunk 4 above)
Comment (gabriellsh) [reply apps/meteor/server/models/raw/Rooms.js:65]: I accidentaly forgot to add this file in the commit. Thanks!

**Review by hugocostadev — APPROVED** [review]

**Review by LucianoPierdona — APPROVED** [review]

**Review by gabriellsh — APPROVED** [review]

**Review by tassoevan — APPROVED** [review]

**Pull request merged**